### PR TITLE
EVG-15321 modify delete user route to require resource ID

### PR DESCRIPTION
**Pull request by ablack12**
[EVG-15321](https://jira.mongodb.org/browse/EVG-15321)

### Description 
We made this route for Mana but they were told that we would accept a resource ID, and apparently we don't, which means that no matter what we're deleting a ton of resources for users. We should only delete roles relevant to the resource, and we should ignore the basic roles so that the user can remain a basic user of Evergreen.

### Testing 
Fixed unit test.

### CODE REVIEW

File: rest/route/user.go [hunk 1]
```diff
@@ -245,27 +257,33 @@ func (h *userPermissionsDeleteHandler) Run(ctx context.Context) gimlet.Responder
...
+		}
+	}
+	if len(rolesToCheck) == 0 {
+		gimlet.NewJSONResponse(struct{}{})
 	}
-	applicableScopes, err := h.rm.FilterScopesByResourceType(scopesIds, h.resourceType)
+
+	rolesForResource, err := h.rm.FilterForResource(rolesToCheck, h.resourceId, h.resourceType)
 	if err != nil {
-		grip.Error(message.WrapError(err, message.Fields{
```
Comment (hadjri) [inline rest/route/user.go:254]: Why are we no longer logging to Splunk here?

File: rest/route/user.go [hunk 2]
```diff
@@ -245,27 +257,33 @@ func (h *userPermissionsDeleteHandler) Run(ctx context.Context) gimlet.Responder
...
-		scopesIds = append(scopesIds, role.Scope)
+	rolesToCheck := []gimlet.Role{}
+	// don't remove basic access, just special access
+	for _, r := range roles {
+		if !utility.StringSliceContains(evergreen.BasicAccessRoles, r.ID) {
+			rolesToCheck = append(rolesToCheck, r)
+		}
+	}
+	if len(rolesToCheck) == 0 {
+		gimlet.NewJSONResponse(struct{}{})
```
Comment (hadjri) [inline rest/route/user.go:268]: What does this line do? Do we need to specify a return statement here?

**Review by hadjri — COMMENTED** [review]
LGTM outside of a couple clarifying questions.

File: rest/route/user.go (hunk 1 above)
Comment (ablack12) [reply rest/route/user.go:254]: It's a bit redundant; we already log errors from rest routes.

File: rest/route/user.go (hunk 2 above)
Comment (ablack12) [reply rest/route/user.go:268]: This just returns an empty successful response; we also do this at 294 at the end of the function but in this case if there are no roles to check we can skip the next query. We should probably also do this in the update function.

**Review by hadjri — APPROVED** [review]

**Pull request merged**